tree_node.go

```diff
@@ -176,9 +179,70 @@ func (node *TreeNode) Rollback(targetVersion Version) bool {
 	}
 	node.Versions = node.Versions[:i+1]
 	node.computeInternalHash()
+	node.nodeSize = uint64(hashSize*len(node.Internals)) + uint64(hashSize*len(node.Versions))
 	return next
 }
 
+func (node *TreeNode) size(oldestVersion Version) (uint64, uint64) {
+	if node == nil {
+		return 0, 0
+	}
+	releasableSize := uint64(0)
```

Comment:
if `node[length-1].version < oldestVersion` then `size == eleasableSize`, anyway to simplify the calculation?